pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -94,3 +96,49 @@ func validateNamespace(namespace string, client Client, fieldPath *field.Path) f
 
 	return allErrs
 }
+
+func validateHcoFeatureGates(client Client, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	hcUnstructured, err := client.GetHyperConverged(context.Background(), "kubevirt-hyperconverged", "openshift-cnv")
```

Comment:
Is it possible that the client would return the "HyperConverged" type in order to prevent the following parsing?
(https://github.com/kubevirt/hyperconverged-cluster-operator/blob/314e87d50b35be7f611964bbcabf16adb9713cf0/pkg/apis/hco/v1beta1/hyperconverged_types.go#L187)

Reply:
I prefer not to use it. I do not see any benefit from using the type. On the contrary. It leads to many dependencies. This means also a lot to add to go.mod.
This also means that any changes in the type will require changes in our code.
In my professional/personal opinion, using the whole type when u need just a subset of it is leads to more trouble than necessary. Extracting just the desired piece of of data makes the code resilient to changes and future versions.